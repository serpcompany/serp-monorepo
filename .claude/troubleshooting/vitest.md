https://vitest.dev/guide/common-errors.html