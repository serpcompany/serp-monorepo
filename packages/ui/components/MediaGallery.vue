<script setup lang="ts">
import type { Company } from '@serp/types/types'

defineProps<{
  company: Company
}>()
</script>

<template>
  <section class="flex w-full flex-col gap-2 md:max-w-[444px]">
    <div class="aspect-video">
      <div
        class="relative flex h-full w-full cursor-pointer items-center justify-center"
      >
        <LazyNuxtImg
          v-if="company?.screenshots && company.screenshots.length"
          :src="company?.screenshots[0]"
          alt="Main Image"
          class="h-full w-full cursor-pointer object-cover"
        />
      </div>
    </div>
    <div class="grid grid-cols-4 gap-2">
      <div v-for="screenshot in company?.screenshots" :key="screenshot">
        <div class="flex h-full w-full items-center justify-center">
          <NuxtImg
            :src="screenshot"
            :alt="`${company.name} screenshots`"
            class="h-full w-full object-contain"
          />
        </div>
      </div>
    </div>
  </section>
</template>
