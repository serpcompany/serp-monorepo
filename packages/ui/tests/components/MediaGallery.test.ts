import { describe, expect, it } from 'vitest'
import MediaGallery from '../../components/MediaGallery.vue'
import ComponentRender from '../componentRender'

describe('MediaGallery Snapshot', () => {
  const scenarios: [string, { props: unknown }][] = [
    [
      'with no screenshots',
      { props: { company: { name: 'Acme Corp', screenshots: [] } } }
    ],
    [
      'with a single screenshot',
      {
        props: {
          company: {
            name: 'Beta Inc',
            screenshots: ['https://example.com/screenshot1.jpg']
          }
        }
      }
    ],
    [
      'with multiple screenshots',
      {
        props: {
          company: {
            name: 'Gamma LLC',
            screenshots: [
              'https://example.com/screenshot1.jpg',
              'https://example.com/screenshot2.jpg',
              'https://example.com/screenshot3.jpg'
            ]
          }
        }
      }
    ]
  ]

  it.each(scenarios)('renders %s correctly', async (desc: string, options: { props: unknown }) => {
    const html = await ComponentRender(`MediaGallery ${desc}`, options, MediaGallery)
    expect(html).toMatchSnapshot()
  })
})
