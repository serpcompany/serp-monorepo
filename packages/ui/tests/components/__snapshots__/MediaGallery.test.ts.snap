// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MediaGallery Snapshot > renders with a single screenshot correctly 1`] = `
<section class="
  flex
  flex-col
  gap-2
  md:max-w-[444px]
  w-full
">
  <div class="aspect-video">
    <div class="
      cursor-pointer
      flex
      h-full
      items-center
      justify-center
      relative
      w-full
    ">
      <img
        alt="Main Image"
        class="
          cursor-pointer
          h-full
          object-cover
          w-full
        "
        data-nuxt-img=""
        src="https://example.com/screenshot1.jpg"
        srcset="https://example.com/screenshot1.jpg 1x, https://example.com/screenshot1.jpg 2x"
      />
    </div>
  </div>
  <div class="
    gap-2
    grid
    grid-cols-4
  ">
    <div>
      <div class="
        flex
        h-full
        items-center
        justify-center
        w-full
      ">
        <img
          alt="Beta Inc screenshots"
          class="
            h-full
            object-contain
            w-full
          "
          data-nuxt-img=""
          src="https://example.com/screenshot1.jpg"
          srcset="https://example.com/screenshot1.jpg 1x, https://example.com/screenshot1.jpg 2x"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`MediaGallery Snapshot > renders with multiple screenshots correctly 1`] = `
<section class="
  flex
  flex-col
  gap-2
  md:max-w-[444px]
  w-full
">
  <div class="aspect-video">
    <div class="
      cursor-pointer
      flex
      h-full
      items-center
      justify-center
      relative
      w-full
    ">
      <img
        alt="Main Image"
        class="
          cursor-pointer
          h-full
          object-cover
          w-full
        "
        data-nuxt-img=""
        src="https://example.com/screenshot1.jpg"
        srcset="https://example.com/screenshot1.jpg 1x, https://example.com/screenshot1.jpg 2x"
      />
    </div>
  </div>
  <div class="
    gap-2
    grid
    grid-cols-4
  ">
    <div>
      <div class="
        flex
        h-full
        items-center
        justify-center
        w-full
      ">
        <img
          alt="Gamma LLC screenshots"
          class="
            h-full
            object-contain
            w-full
          "
          data-nuxt-img=""
          src="https://example.com/screenshot1.jpg"
          srcset="https://example.com/screenshot1.jpg 1x, https://example.com/screenshot1.jpg 2x"
        />
      </div>
    </div>
    <div>
      <div class="
        flex
        h-full
        items-center
        justify-center
        w-full
      ">
        <img
          alt="Gamma LLC screenshots"
          class="
            h-full
            object-contain
            w-full
          "
          data-nuxt-img=""
          src="https://example.com/screenshot2.jpg"
          srcset="https://example.com/screenshot2.jpg 1x, https://example.com/screenshot2.jpg 2x"
        />
      </div>
    </div>
    <div>
      <div class="
        flex
        h-full
        items-center
        justify-center
        w-full
      ">
        <img
          alt="Gamma LLC screenshots"
          class="
            h-full
            object-contain
            w-full
          "
          data-nuxt-img=""
          src="https://example.com/screenshot3.jpg"
          srcset="https://example.com/screenshot3.jpg 1x, https://example.com/screenshot3.jpg 2x"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`MediaGallery Snapshot > renders with no screenshots correctly 1`] = `
<section class="
  flex
  flex-col
  gap-2
  md:max-w-[444px]
  w-full
">
  <div class="aspect-video">
    <div class="
      cursor-pointer
      flex
      h-full
      items-center
      justify-center
      relative
      w-full
    ">
      <!-- v-if -->
    </div>
  </div>
  <div class="
    gap-2
    grid
    grid-cols-4
  "></div>
</section>
`;
